sds/src/scoped_ruleset/mod.rs

```diff
@@ -190,6 +195,10 @@ where
                 });
             }
         }
+
+        // Sanitize the segment and push it
+        self.sanitized_segments_until_node.push(segment.sanitize());
```

Comment:
You are pushing to `sanitized_segments_until_node`, but never popping. I think this needs a `.pop` in `.pop_segment`?

Reply:
Yes my bad, I'm adding the pop part

Reply:
[Pop sanitized_segments_until_node when popping a segment](https://github.com/DataDog/dd-sensitive-data-scanner/pull/96/commits/61bd7347f48efd240d84f60924c60d70ba4febc2)